Subscribers/ProductStreams.php

```diff
@@ -28,14 +59,53 @@ public function extendBackendProductStream(\Enlight_Event_EventArgs $args)
         $subject = $args->getSubject();
         $request = $subject->Request();
 
-        switch($request->getActionName()) {
+        switch ($request->getActionName()) {
+            case 'load':
+                $this->registerMyTemplateDir();
+                $this->registerMySnippets();
+
+                $subject->View()->extendsTemplate(
+                    'backend/product_stream/view/selected_list/connect_product.js'
+                );
+                break;
             case 'delete':
                 $streamId = $request->get('id');
 
                 if ($this->getProductStreamService()->isStreamExported($streamId)) {
                     $this->getSDK()->recordStreamDelete($streamId);
                 }
                 break;
+            case 'addSelectedProduct':
+                $streamId = $request->getParam('streamId');
+                $articleId = $request->getParam('articleId');
+
+                if (!$this->getProductStreamService()->isStreamExported($streamId)) {
+                    return;
+                }
+
+                $autoUpdate = $this->config->getConfig('autoUpdateProducts', 1);
```

Comment:
if autoUpdateProducts has value 2 that means products are update via cronjob. So if statement should be
```
 if ($autoUpdate == 0) {
     return;
 }
```
after that you already have 
```
if ($autoUpdate == 1) {
...
} elseif ($autoUpdate == 2) {
...
}
```